python/cudf/cudf/_lib/column.pyx

```diff
@@ -428,6 +438,47 @@ cdef class Column:
             offset,
             children)
 
+    cpdef pylibcudf.Column to_pylibcudf(self):
+        # TODO: Categoricals will need to be treated differently eventually.
+        # There is no 1-1 correspondence between cudf and libcudf for
+        # categoricals due to the ordering question.
+        if is_categorical_dtype(self.dtype):
+            col = self.base_children[0]
+        else:
+            col = self
+
+        cdef pylibcudf.DataType dtype = dtype_to_pylibcudf_type(col.dtype)
+
+        cdef pylibcudf.gpumemoryview data = None
+        if col.base_data is not None:
+            data = pylibcudf.gpumemoryview(col.base_data)
+
+        cdef pylibcudf.gpumemoryview mask = None
+        if self.nullable:
+            # TODO: Are we intentionally use self's mask instead of col's?
+            # Where is the mask stored for categoricals?
+            mask = pylibcudf.gpumemoryview(self.base_mask)
+
+        cdef Column child_column
+        children = []
+        if col.base_children:
+            for child_column in col.base_children:
+                children.append(child_column.to_pylibcudf())
+
+        return pylibcudf.Column(
```

Comment:
Presume this method is cheap enough that we don't really care about memoizing.

Reply:
I'd say let's put off memoizing for now since as you brought up in your comment above the long term vision is for a cudf's columns to own pylibcudf columns, and as such there won't be "conversion" in the long run so much as just accessing the owned object.